src/source-updater.js

```diff
@@ -280,7 +280,13 @@ const actions = {
 
     sourceUpdater.logger_(`changing ${type}Buffer codec from ${sourceUpdater.codecs[type]} to ${codec}`);
 
-    sourceBuffer.changeType(mime);
+    // check if change to the provided type is supported
+    try {
+      sourceBuffer.changeType(mime);
+    } catch (e) {
+      videojs.log.warn(e);
+    }
+
     sourceUpdater.codecs[type] = codec;
```

Comment:
```suggestion
    // check if change to the provided type is supported
    try {
      sourceBuffer.changeType(mime);
      sourceUpdater.codecs[type] = codec;
    } catch (e) {
      videojs.log.warn(`Failed to changeType on ${type}Buffer`, e);
    }
```

Reply:
Hi @dzianis-dashkevich

thank you for your reply and suggestion.

Have you tried your changes? Is it working now as expected? I definitely gonna try this to see the behaviour. 

Thanks again!